internal/ingest/linksystem.go

```diff
@@ -345,86 +324,61 @@ func (i *legIngester) processEntries(adCid cid.Cid, p peer.ID, nentries ipld.Nod
 	return nil
 }
 
-// batchIndexerEntries starts a goroutine that reads multihashes from two input
-// channels, putChan and removeChan.  The goroutine collects these into
-// separate slices, storing up to batchSize elements.  When a slice is at
-// capacity, a Put or Remove request is made to the indexer core.  This
-// function returns an error channel that returns an error if one occurs during
-// Put or Remove, which also indicates the goroutine has exited (and will no
-// longer read its input channels).
+// batchIndexerEntries starts a goroutine that processes batches of multihashes
+// from an input channels.  The goroutine collects these into a slice, storing
+// up to batchSize elements.  When the slice is at capacity, a Put or Remove
+// request is made to the indexer core depending on the whether isRm is true
+// or false.  This function returns an error channel that returns an error if
+// one occurs during processing.  This also indicates the goroutine has exited
+// (and will no longer read its input channel).
 //
-// The goroutine exits when both the input channels are closed.  It closes the
-// error channel to indicate completion.
-func batchIndexerEntries(batchSize int, putChan, removeChan <-chan multihash.Multihash, value indexer.Value, idxr indexer.Interface) <-chan error {
+// The goroutine exits when the input channel is closed.  It closes the error
+// channel to indicate completion.
+func (li *legIngester) batchIndexerEntries(mhChan <-chan multihash.Multihash, value indexer.Value, isRm bool) <-chan error {
+	var indexFunc func(indexer.Value, ...multihash.Multihash) error
+	var opName string
+	if isRm {
+		indexFunc = li.indexer.Remove
+		opName = "remove"
```

Comment:
can we have constant / enum for these rather than strings?

Reply:
It is only for a log message, here only, so no.